### ISSUE TITLE
Accept integers for 1D shapes

### ISSUE THREAD

**Issue by Shians** [h5py/h5py#1229]
I am trying to manually set the chunking options in a 1D array. But the obvious ways to do this fail.

```python
import h5py
import numpy as np

data = np.random.randint(0, 1000, 1000000, dtype = "i2")

f = h5py.File("array_data.hdf5", "w")
f.create_dataset("data", data = data, dtype = "i2", chunks = 50000)
f.close()
```

This gives error:

```python
>>> f.create_dataset("data", data = data, dtype = "i2", chunks = 50000)
Traceback (most recent call last):
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 129, in rq_tuple
    tpl = tuple(tpl)
TypeError: 'int' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/group.py", line 136, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/dataset.py", line 140, in make_new_dset
    maxshape, scaleoffset, external)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 135, in fill_dcpl
    rq_tuple(chunks, 'chunks')
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 131, in rq_tuple
    raise TypeError('"%s" argument must be None or a sequence object' % name)
TypeError: "chunks" argument must be None or a sequence object
```

Alternatively making the chunks a tuple of 1

```python
f = h5py.File("array_data.hdf5", "w")
f.create_dataset("data", data = data, dtype = "i2", chunks = (50000))
f.close()
```

fails due to 

```python
>>> f.create_dataset("data", data = data, dtype = "i2", chunks = (50000))
Traceback (most recent call last):
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 129, in rq_tuple
    tpl = tuple(tpl)
TypeError: 'int' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/group.py", line 136, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/dataset.py", line 140, in make_new_dset
    maxshape, scaleoffset, external)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 135, in fill_dcpl
    rq_tuple(chunks, 'chunks')
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 131, in rq_tuple
    raise TypeError('"%s" argument must be None or a sequence object' % name)
TypeError: "chunks" argument must be None or a sequence object
```

So is it possible to set chunks for 1D arrays?

 * Operating System : MacOS 10.11
 * Python version: 3.7.3
 * Python acquired: Anaconda on MacOS
 * h5py version: 2.9.0
 * HDF5 version: 1.10.4

**Comment by takluyver** [comment]
`(50000)` doesn't make a tuple - it's still just an integer with parentheses. You need a comma: `(50000,)`.

**Comment by takluyver** [comment]
Although I think it makes sense for h5py to accept an integer for a 1D chunk shape. Looking at a few functions in numpy, they all allow "int or tuple of ints" for the shape parameter. So I'll leave this issue open as a feature request.

**Comment by takluyver** [comment]
Fixed by #1340.
